internal/consensus/state.go

```diff
@@ -1408,21 +1407,9 @@ func (cs *State) enterPrecommit(height int64, round int32) {
 		panic(fmt.Sprintf("this POLRound should be %v but got %v", round, polRound))
 	}
 
-	// +2/3 prevoted nil. Unlock and precommit nil.
+	// +2/3 prevoted nil. Precommit nil.
```

Comment:
Side comment: we should probably have a helper function that wraps `len(blockID.Hash) == 0`, for example `isVoteNil` or something like that, as code would read better.

Reply:
Also I am wondering what exactly the check at line 1405 brings? I don't see from correctness perspective entering precommit in case there is a polka for higher round. As this also depends on the way gossip is implemented and skipping round logic, we could end up in scenario where this condition is not true and then we would panic. As a more general rule of thumb, I think we should have clear description or even better invariant violation whenever we panic in the code.

Reply:
There's a helper for nil block checking. I just renamed it now to `blockID.IsNil()` and updated the code in `state.go` to use it.